src/execution/runner.rs

```diff
@@ -0,0 +1,240 @@
+use std::{
+    cell::RefCell,
+    rc::Rc,
+    sync::atomic::{AtomicBool, Ordering},
+    time::Duration,
+};
+
+use pyo3::{Py, PyResult, Python};
+use timely::{
+    communication::Allocate,
+    dataflow::{
+        operators::{Filter, Probe},
+        ProbeHandle,
+    },
+    progress::Timestamp,
+    worker::Worker,
+};
+
+use crate::{
+    dataflow::Dataflow,
+    errors::PythonException,
+    inputs::EpochInterval,
+    recovery::{
+        model::{
+            FlowStateBytes, KChange, ProgressReader, ResumeEpoch, ResumeFrom, StateReader, StoreKey,
+        },
+        operators::{read, BroadcastWrite, Recover, Summary, Write},
+        python::{RecoveryBuilder, RecoveryConfig},
+        store::in_mem::{InMemProgress, InMemStore, StoreSummary},
+    },
+};
+
+use super::{build_production_dataflow, PeriodicSpan, WorkerCount, WorkerIndex};
+
+pub(crate) struct WorkerRunner<'a, A: Allocate> {
```

Comment:
Perhaps call this module `worker`? This is where we define the specifics of the behavior or a worker thread and would better contrast it with the `run` module.

I might also go as far as calling this just `bytewax:worker::Worker` to contrast `timely::worker::Worker`, but I could go either way.

Reply:
I like `bytewax::worker::Worker`. It shouldn't cause too much confusion since at that point we only use `timely::worker::Worker` in that module, so I moved the `WorkerRunner` and Worker related stuff inside `bytewax::worker`, and renamed `WorkerRunner` to `Worker`.